src/Domain/DoctrineDonationRepository.php

```diff
@@ -90,10 +87,28 @@ public function allocateMatchFunds(Donation $donation): string
             $amountNewlyMatched = bcadd($amountNewlyMatched, $newWithdrawalAmount, 2);
         }
 
-        $this->getEntityManager()->flush(); // Flush `$newWithdrawals` if any.
+        try {
+            // Flush `$newWithdrawals` if any and updates to DB copies of CampaignFundings.
+            $this->getEntityManager()->flush();
+        } catch (\Throwable $exception) {
+            $this->getLogger()->error(sprintf(
+                'Failed to flush DB for new withdrawals so donation UUID %s did not get matching: %s',
+                $donation->getUuid()->toString(),
+                $exception->getMessage(),
+            ));
+
+            // todo pick one
+            $this->matchingAdapter->releaseNewlyAllocatedFunds();
+            //$this->reverseMatchingAllocations($donation, $newWithdrawals, $likelyAvailableFunds);
+
+            // Ensure nothing later tries to persist the pending Donation or CampaignFunding changes.
+            $this->getEntityManager()->clear();
+
+            return '0.0';
```

Comment:
definitely think we shouldn't return after catching any Throwable - we don't know what went wrong so safest thing we can do is re-throw.